chiselFrontend/src/main/scala/chisel3/core/Bits.scala

```diff
@@ -550,12 +550,20 @@ private[core] sealed trait UIntFactory {
   // NOTE: This resolves UInt(width = 32)
   def apply(dir: Option[Direction] = None, width: Int): UInt = apply(Width(width))
   /** Create a UInt literal with inferred width.- compatibility with Chisel2. */
-  def apply(value: BigInt): UInt = apply(value, Width())
+  def apply(value: BigInt)(implicit opts: CompileOptions): UInt = {
+    if (opts.deprecateSingleArgumentFactoryMethods) {
```

Comment:
Might it make sense to factor this into a Builder method that takes an implicit CompileOptions?
So that these three lines would just be:
```
Builder.deprecatedSingleArgumentFactoryMethods("helpful message here")
```
(and have the `if` test be inside the method)